### ISSUE TITLE
Conversion between Unicode codepoint values and chars

### ISSUE THREAD

**Issue by epbuennig** [typst/typst#1068]
### Description

Add a function that allows converting between code point values. A free function has the advantage of being simple to use for single chars and methods like `codepoints`, `clusters`. A single function can fill the role of both conversions, but that leaves the possibility of some confusing bugs when types don't match.

```rust
// à
#let test = "a\u{0300}"

#test.codepoints() == ("a", "\u{300}")
#test.codepoints().map(unicode) == (97, 768)

// is this a horrible idea? maybe
#unicode("a") == 97
#unicode(97) == "a"
```

Alternatives that were considered were a flag on `codepoints` and `clusters` to return ints/arrays of ints instead.
```rust
#test.codepoints(values: true) == ("a", "\u{300}")
#test.clusters(values: true) == (("a", "\u{300}"),)
```
But that leaves the use case for a single char being awkward and the `int -> str` conversion unsolved.

### Use Case

Converting between either is currently unwieldy or down right impossible without external tools or manual labor which is prone to errors.

**Comment by epbuennig** [comment]
Related: https://github.com/typst/typst/issues/918

**Comment by epbuennig** [comment]
If https://github.com/typst/typst/pull/1032 lands, then this could be done using `str.unicode` and `int.unicode` which already solves the issue of unicode possibly causing hard to debug bugs in cases where types are very dynamic.

**Comment by epbuennig** [comment]
Closed by #1132
